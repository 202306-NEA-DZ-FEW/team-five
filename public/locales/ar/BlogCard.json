{
  "BlogCard": {
    "Read More": "إقرأ المزيد"
  }
  
}
