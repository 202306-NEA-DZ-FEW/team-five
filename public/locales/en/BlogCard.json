{
  "BlogCard": {
    "Read More": "Read More"
  }
  
}
