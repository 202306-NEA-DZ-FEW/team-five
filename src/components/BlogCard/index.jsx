import Link from "next/link";
import { useTranslation } from "next-i18next";

const BlogCard = ({ blog }) => {
    const { t } = useTranslation("BlogCard");

    const authorsNames = [
        "Achraf Mehanneche ",
        "Sidahmed Slikh    ",
        "Mohammed Tchoketch",
        "Abdelhamid Serhan ",
        "Mellisa Sidisaid  ",
    ];
    const images = [
        "achraf.jpg",
        "sidahmed.jpg",
        "admin.svg",
        "hamid_ahmidat.jpg",
        "melisa.jpg",
    ];

    const randomIndex = Math.floor(Math.random() * authorsNames.length);
    const randomAuthorName = authorsNames[randomIndex];
    const randomAuthorImage = images[randomIndex];

    const truncateText = (text, maxLength) => {
        if (text.length <= maxLength) {
            return text;
        }
        return text.slice(0, maxLength) + "... ";
    };

    const truncatedTitle = truncateText(blog.title, 65);
    const truncatedDescription = truncateText(blog.description, 210);

    return (
        <div className='relative w-[416px] h-[528px] rounded overflow-hidden shadow-lg m-4 hover:bg-gray-100 transform hover:scale-105 transition duration-300 ease-in-out'>
            <img
                src={blog.image}
                alt={blog.title}
                className='w-full h-56 object-cover'
            />
            <div className='flex flex-col flex-grow p-4'>
                <h2 className='font-bold text-xl mb-2'>{truncatedTitle}</h2>
                <div className='flex items-center mb-2'>
                    <img
                        src={`./${randomAuthorImage}`}
                        alt={randomAuthorName}
                        className='w-8 h-8 rounded-full mr-2'
                    />
                    <div className='flex space-x-4 mr-4 text-gray-700'>
                        {randomAuthorName}
                    </div>
                    <div className='flex space-x-4 text-gray-700'>
                        {blog.publishedAt}
                    </div>
                </div>
                <p className='text-gray-600 mb-4 flex-grow p-4'>
                    {truncatedDescription}
                </p>
                <div className='mt-auto'></div>

                <button
                    className='absolute bottom-0 left-0 hover:text-gray-300 text-green-500 font-bold py-2 px-4 mb-1 rounded-full text-center'
                    style={{ textDecoration: "underline" }}
                >
                    <p>{t("Read")}</p>
                </button>
            </div>
        </div>
    );
};

export default BlogCard;
