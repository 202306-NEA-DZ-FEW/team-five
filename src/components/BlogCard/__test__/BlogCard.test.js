import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

import BlogCard from "..";

const mockRandomIndex = jest.spyOn(Math, "floor");
mockRandomIndex.mockReturnValue(0);

jest.mock("next/router", () => ({
    useRouter: () => ({
        route: "/",
        pathname: "",
        query: {},
        asPath: "",
    }),
}));

jest.mock("react-i18next", () => ({
    useTranslation: () => ({ t: (key) => key, i18n: { language: "en" } }),
}));

const mockStore = configureMockStore([]);
const store = mockStore({
    userInfo: null,
});

describe("Blog card", () => {
    it("should render correctly", () => {
        const blog = {
            title: "Sample Title",
            description: "Sample Description",
            image: "sample-image.jpg",
            publishedAt: "2023-11-07",
            author: "Mellisa Sidisaid",
            authorImage: "melisa.jpg",
        };

        const mockStore = configureMockStore([]);
        const store = mockStore({
            userInfo: null,
        });

        const component = renderer.create(
            <Provider store={store}>
                <BlogCard blog={blog} />
            </Provider>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
