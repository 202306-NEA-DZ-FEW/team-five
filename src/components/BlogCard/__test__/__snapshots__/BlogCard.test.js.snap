// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard Component should render correctly 1`] = `
<div
  className="relative w-[416px] h-[528px] rounded overflow-hidden shadow-lg m-4 hover:bg-gray-100 transform hover:scale-105 transition duration-300 ease-in-out"
>
  <img
    alt=""
    className="w-full h-56 object-cover"
    src=""
  />
  <div
    className="flex flex-col flex-grow p-4"
  >
    <h2
      className="font-bold text-xl mb-2"
    />
    <div
      className="flex items-center mb-2"
    >
      <img
        alt="Mellisa Sidisaid"
        className="w-8 h-8 rounded-full mr-2"
        src="./melisa.jpg"
      />
      <div
        className="flex space-x-4 mr-4 text-gray-700"
      >
        Mellisa Sidisaid
      </div>
      <div
        className="flex space-x-4 text-gray-700"
      />
    </div>
    <p
      className="text-gray-600 mb-4 flex-grow p-4"
    />
    <div
      className="mt-auto"
    />
    <button
      className="absolute bottom-0 left-0 hover:text-gray-300 text-green-500 font-bold py-2 px-4 mb-1 rounded-full text-center"
      style={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      <p>
        Read
      </p>
    </button>
  </div>
</div>
`;
